modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/ZonePartitionReplicaListener.java

```diff
@@ -207,7 +212,15 @@ private static PendingTxPartitionEnlistment createAbandonedTxRecoveryEnlistment(
     public CompletableFuture<ReplicaResult> invoke(ReplicaRequest request, UUID senderId) {
         return replicaPrimacyEngine.validatePrimacy(request)
                 .thenCompose(replicaPrimacy -> processRequest(request, replicaPrimacy, senderId))
-                .thenApply(res -> {
+                .handle((res, ex) -> {
+                    if (ex != null) {
+                        if (hasCause(ex, LockException.class)) {
+                            sneakyThrow(new OperationLockException(request.getClass(), (LockException) unwrapCause(ex)));
```

Comment:
I do not think a request class is enough to identify an operation because we have several operations in two request classes: ReadWriteMultiRowReplicaRequest and ReadWriteSingleRowReplicaRequest.
I rather think about SingleRowReplicaRequest#requestType.